homeassistant/components/dynalite/config_flow.py

```diff
@@ -18,19 +21,68 @@ def __init__(self) -> None:
         """Initialize the Dynalite flow."""
         self.host = None
 
+    def existing_host_entry(self, config) -> Any:
+        """Check if a specific host is already configured and return its entry or None."""
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.data[CONF_HOST] == config[CONF_HOST]:
+                return entry
+        return None
+
+    async def check_bridge_connection(self, config) -> bool:
+        """Check whether this bridge is available to connect."""
+        bridge = DynaliteBridge(self.hass, config)
+        if not await bridge.async_setup():
+            LOGGER.error("Unable to setup bridge - config=%s", config)
+            return False
+        return True
+
     async def async_step_import(self, import_info: Dict[str, Any]) -> Any:
         """Import a new bridge as a config entry."""
         LOGGER.debug("Starting async_step_import - %s", import_info)
         host = import_info[CONF_HOST]
-        for entry in self.hass.config_entries.async_entries(DOMAIN):
-            if entry.data[CONF_HOST] == host:
-                if entry.data != import_info:
-                    self.hass.config_entries.async_update_entry(entry, data=import_info)
-                return self.async_abort(reason="already_configured")
+        entry = self.existing_host_entry(import_info)
+        if entry:
+            if entry.data != import_info:
+                self.hass.config_entries.async_update_entry(entry, data=import_info)
+            return self.async_abort(reason="already_configured")
         # New entry
-        bridge = DynaliteBridge(self.hass, import_info)
-        if not await bridge.async_setup():
+        if not await self.check_bridge_connection(import_info):
             LOGGER.error("Unable to setup bridge - import info=%s", import_info)
-            return self.async_abort(reason="no_connection")
+            return self.async_abort(reason="cannot_connect")
         LOGGER.debug("Creating entry for the bridge - %s", import_info)
         return self.async_create_entry(title=host, data=import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        # This is for backwards compatibility.
```

Comment:
Please remove the init step and just use the user step. There's no backwards compatibility to handle.

Reply:
done